### ISSUE TITLE
[BUDI-7798] Allow relationships within the same table

### ISSUE THREAD

**Issue by marco-santulli** [Budibase/budibase#3498]
creating a relationship over the same table is a valid/frequent use case.  Imagine a contractor/subcontractor relationship.  They are both contractors.

I have seen this implemented differently in other products (directus / nocodb), it might be good to allow the developer to use columns in a third table provide the mapping between the two tables.  Currently I think only a foreign key column in the end-table is allowed.

This is the current user experience, the same happens in the budibase DB and external sources:

![image](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/68911bf8-7f21-4551-9376-95b5ab3cdbaa/7f62517d-2195-4494-be56-93d731d089fe?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC82ODkxMWJmOC03ZjIxLTQ1NTEtOTM3Ni05NWI1YWIzY2RiYWEvN2Y2MjUxN2QtMjE5NS00NDk0LWJlNTYtOTNkNzMxZDA4OWZlIiwiaWF0IjoxNzAxNDQ2ODcxLCJleHAiOjE3MDE1MzMyNzF9.N7D6J6GbmM13jVUNJr1VM7D1_iN7Ee6pks8vOvLrnOY)

<sub>From [SyncLinear.com](https://synclinear.com) | [BUDI-7798](https://linear.app/budibase/issue/BUDI-7798/allow-relationships-within-the-same-table)</sub>

**Comment by santossg** [comment]
Is this solved? I want to use a relationship within the same table.
Please let me know if there is a silution.

**Comment by arpadgabor** [comment]
Any chance of this getting some love?

**Comment by pmankow** [comment]
This is necessary. I want it as well.

**Comment by arpadgabor** [comment]
Hi, I totally understand the need to close issues that are not planned and I also understand there probably are more important problems to solve. However I'd like to ask, if possible, why is the feature not planned? (complexity, few requests?) or maybe we can get an answer about any workarounds we don't know of?

**Comment by pmankow** [comment]
I agree. I really don't understand how it can be avoided. At least it would be nice if you could provide a functional/logical workaround or note on how implementing this violates the design philosophy of the product.

**Comment by melohagan** [comment]
Sorry closed by accident!

**Comment by leolivier** [comment]
I need that too!

**Comment by L42xyz** [comment]
I need that as well. I have a contact table where people can be sponsor of other people and I am currently stuck. Thanks in advance
